### PR TITLE
Fix locks used in MockConsumer [13303]

### PR DESCRIPTION
**Pull request by MiguelCompany**
The class used on unit tests for checking behavior regarding the log messages is wrongly used.
A reference to its condition_variable is obtained and then used with a different mutex.

This PR fixes this by adding a generic `wait` method receiving a predicate, as long as a `wait_for_at_least_entries` method, which is the most common use in the unit tests.

The `cv()` getter and `ConsumedEntriesSize_nts()` methods are removed, to ensure we don't have incorrect behaviour.

### CODE REVIEW

File: test/unittest/statistics/dds/StatisticsDomainParticipantMockTests.cpp [hunk 1]
```diff
@@ -42,11 +42,7 @@ class StatisticsDomainParticipantMockTests : public ::testing::Test
     void helper_block_for_at_least_entries(
             uint32_t amount)
     {
-        std::unique_lock<std::mutex> lck(*mutex_);
```
Comment (richiware) [inline test/unittest/statistics/dds/StatisticsDomainParticipantMockTests.cpp:45]: I think this mutex is only used here. Remove it.

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 2]
```diff
@@ -136,11 +136,7 @@ class StatisticsDomainParticipantTests : public ::testing::Test
     void helper_block_for_at_least_entries(
             uint32_t amount)
     {
-        std::unique_lock<std::mutex> lck(*mutex_);
```
Comment (richiware) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:139]: I think this mutex is only used here. Remove it.

File: test/unittest/xmlparser/XMLElementParserTests.cpp [hunk 3]
```diff
@@ -61,11 +61,7 @@ class XMLParserTests : public ::testing::Test
     void helper_block_for_at_least_entries(
             uint32_t amount)
     {
-        std::unique_lock<std::mutex> lck(*xml_mutex_);
```
Comment (richiware) [inline test/unittest/xmlparser/XMLElementParserTests.cpp:64]: I think this mutex is only used here. Remove it.

File: test/unittest/xmlparser/XMLEndpointParserTests.cpp [hunk 4]
```diff
@@ -59,11 +59,7 @@ class XMLEndpointParserTests : public ::testing::Test
     void helper_block_for_at_least_entries(
             uint32_t amount)
     {
-        std::unique_lock<std::mutex> lck(*xml_mutex_);
```
Comment (richiware) [inline test/unittest/xmlparser/XMLEndpointParserTests.cpp:62]: I think this mutex is only used here. Remove it.

File: test/unittest/xmlparser/XMLParserTests.hpp [hunk 5]
```diff
@@ -68,11 +68,7 @@ class XMLParserTests : public ::testing::Test
     void helper_block_for_at_least_entries(
             uint32_t amount)
     {
-        std::unique_lock<std::mutex> lck(*xml_mutex_);
```
Comment (richiware) [inline test/unittest/xmlparser/XMLParserTests.hpp:71]: I think this mutex is only used here. Remove it.

**Review by richiware — CHANGES_REQUESTED** [review]

File: test/unittest/statistics/dds/StatisticsDomainParticipantMockTests.cpp (hunk 1 above)
Comment (MiguelCompany) [reply test/unittest/statistics/dds/StatisticsDomainParticipantMockTests.cpp:45]: done

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 2 above)
Comment (MiguelCompany) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:139]: done

File: test/unittest/xmlparser/XMLElementParserTests.cpp (hunk 3 above)
Comment (MiguelCompany) [reply test/unittest/xmlparser/XMLElementParserTests.cpp:64]: done

File: test/unittest/xmlparser/XMLEndpointParserTests.cpp (hunk 4 above)
Comment (MiguelCompany) [reply test/unittest/xmlparser/XMLEndpointParserTests.cpp:62]: done

File: test/unittest/xmlparser/XMLParserTests.hpp (hunk 5 above)
Comment (MiguelCompany) [reply test/unittest/xmlparser/XMLParserTests.hpp:71]: done

**Review by richiware — APPROVED** [review]

**Pull request merged**